logprep/framework/pipeline.py

```diff
@@ -358,76 +444,17 @@ def _shut_down(self):
             processor.shut_down()
 
     def _drain_input_queues(self):
-        if not hasattr(self._input, "_messages"):
+        if not hasattr(self._input, "messages"):
             return
-        if isinstance(self._input._messages, queue.Queue):
-            while self._input._messages.qsize():
+        if isinstance(self._input.messages, queue.Queue):
+            while self._input.messages.qsize():
                 self._process_pipeline()
 
     def stop(self):
         """Stop processing processors in the Pipeline."""
         self._continue_iterating = False
 
 
-class SharedCounter:
-    """A shared counter for multi-processing pipelines."""
-
-    CHECKING_PERIOD = 0.5
-
-    def __init__(self):
-        self._val = Value(c_ulonglong, 0)
-        self._lock = Lock()
-        self._timer = Value(c_double, 0)
-        self._checking_timer = 0
-        self._logger = None
-        self._period = None
-
-    def _init_timer(self, period: float):
-        if self._period is None:
-            self._period = period
-        with self._lock:
-            self._timer.value = time() + self._period
-
-    def _create_logger(self, log_handler: Handler):
-        if self._logger is None:
-            logger = Logger("Processing Counter", level=log_handler.level)
-            for handler in logger.handlers:
-                logger.removeHandler(handler)
-            logger.addHandler(log_handler)
-
-            self._logger = logger
-
-    def setup(self, print_processed_period: float, log_handler: Handler):
-        """Setup shared counter for multiprocessing pipeline."""
-        self._create_logger(log_handler)
-        self._init_timer(print_processed_period)
-        self._checking_timer = time() + self.CHECKING_PERIOD
-
-    def increment(self):
-        """Increment the counter."""
-        with self._lock:
-            self._val.value += 1
-
-    def print_if_ready(self):
-        """Periodically print the counter and reset it."""
-        current_time = time()
-        if current_time > self._checking_timer:
-            self._checking_timer = current_time + self.CHECKING_PERIOD
-            if self._timer.value != 0 and current_time >= self._timer.value:
-                with self._lock:
-                    if self._period / 60.0 < 1:
-                        msg = f"Processed events per {self._period} seconds: {self._val.value}"
-                    else:
-                        msg = (
-                            f"Processed events per {self._period / 60.0:.2f} minutes: "
-                            f"{self._val.value}"
-                        )
-                    if self._logger:
-                        self._logger.info(msg)
-                    self._val.value = 0
-                    self._timer.value = time() + self._period
-
-
```

Comment:
It's not part of your changes here, but could you please also fix the wrong type in the MultiprocessingPipeline init?

`used_server_ports: list,` should be `used_server_ports: dict,`.

Reply:
sure -> done